### ISSUE TITLE
precision_recall supplying results in wrong order

### ISSUE THREAD

**Issue by jeweinb** [PyTorchLightning/metrics#494]
Below is an example of how to re-create the problem. If you call a function named precision_recall() you would expect to get results in the order of precision, recall. Torchmetrics returns the numbers in the order of recall, precision.

```python
import torch
import torchmetrics.functional as tmf
from sklearn.metrics import classification_report, precision_recall_fscore_support

x = torch.tensor([1, 2, 1, 0, 4, 3, 2, 1, 2, 3, 4, 4])
y = torch.tensor([1, 0, 1, 0, 2, 3, 2, 1, 1, 3, 2, 4])

print(tmf.f1(x, y, num_classes=5, average='macro'))
# tensor(0.6714)
print(tmf.precision_recall(x, y, num_classes=5, average='macro'))
# (tensor(0.7333), tensor(0.7167))
print(precision_recall_fscore_support(x, y, average='macro'))
# (0.7166666666666666, 0.7333333333333334, 0.6714285714285714, None)

print(classification_report(x, y))
#               precision    recall  f1-score   support
#            0       0.50      1.00      0.67         1
#            1       0.75      1.00      0.86         3
```

**Comment by Borda** [comment]
That is a good point, mind send a PR to fix it?
cc: @SkafteNicki

**Comment by SkafteNicki** [comment]
Sounds good to me :] please feel free to send a PR

**Comment by jeweinb** [comment]
will do, ty

**Comment by SkafteNicki** [comment]
After taking a closer look there is no issue here. The difference is due to we having a different input order than sklearn e.g. `preds` first then `target` (this is how pytorch does it in their loss functions) compared to sklearn which have `target` first and then `preds`.
Thus, if you change your call to `precision_recall_fscore_support` to:
```
print(precision_recall_fscore_support(y, x, average='macro'))  # swap order of x and y
# (0.7333333333333334, 0.7166666666666666, 0.6714285714285714, None)
```
it will be in the same order. Closing issue.
